internal/handlers/common/update.go

```diff
@@ -18,33 +18,60 @@ import (
 	"fmt"
 
 	"github.com/FerretDB/FerretDB/internal/types"
-	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
 // UpdateDocument updates the given document with a series of update operators.
-func UpdateDocument(doc, update *types.Document) error {
+// Returns true if document was changed.
+func UpdateDocument(doc, update *types.Document) (bool, error) {
 	for _, updateOp := range update.Keys() {
 		updateV := must.NotFail(update.Get(updateOp))
 
 		switch updateOp {
 		case "$set":
-			setDoc, err := AssertType[*types.Document](updateV)
-			if err != nil {
-				return err
+			msgFmt := `Modifiers operate on fields but we found type %s instead. ` +
+				`For example: {$mod: {<field>: ...}} not {$set: %s}`
+
+			if updateV == nil ||
```

Comment:
This check could be placed inside switch, isn't it?

Reply:
this is check for the value, not for the type. Can I?

Reply:
after I've changed the code for https://github.com/FerretDB/FerretDB/pull/634#discussion_r881583187 comment, it works in the `switch` case.